xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -324,17 +339,30 @@ bool CPVRClients::IsCreatedClient(int iClientId) const
   return GetCreatedClient(iClientId, client);
 }
 
-bool CPVRClients::IsCreatedClient(const std::string& id) const
+bool CPVRClients::IsCreatedClient(const std::string& addonId, const std::string& instanceId) const
 {
   CSingleLock lock(m_critSection);
   for (const auto& client : m_clientMap)
   {
-    if (client.second->ID() == id)
+    if (client.second->ID() == addonId && client.second->GetUsedInstanceID() == instanceId)
```

Comment:
I would omit the "Used" from the method's name. If it is returned it is obvious that it is used, right?

Reply:
Can be removed and used as `GetInstanceID()`, has done it a bit bigger as on `IAddonInstanceHandler` also a `const std::string& InstanceID()`.

The only problem by IAddonInstanceHandler was that a fallback id is created in the instance class using class pointer (if an empty was given by constructor) , which then makes it unusable for PVR, because the next time Kodi is started it will be different.